utils/experiment_utils.py

```diff
@@ -550,6 +556,9 @@ def load_state_dict(self, state_dict: Dict[str, Any]):
         for ps, stage_info in zip(self.pipeline_stages, state_dict["stage_info_list"]):
             ps.early_stopping_criterion_met = stage_info["early_stopping_criterion_met"]
             ps.steps_taken_in_stage = stage_info["steps_taken_in_stage"]
+            ps.offpolicy_steps_taken_in_stage = stage_info[
+                "offpolicy_steps_taken_in_stage"
+            ]
```

Comment:
Will this break loading checkpoints saved before adding off policy training? Can we instead do `stage_info.get("offpolicy_steps_taken_in_stage", 0)`? This probably applies to `self.off_policy_epochs = state_dict["off_policy_epochs"]` below too.